larray/core/session.py

```diff
@@ -781,6 +780,50 @@ def compact(self, display=False):
             new_items.append((k, compacted))
         return Session(new_items)
 
+    def apply(self, func, *args, **kwargs):
+        """
+        Apply function `func` on all elements of the session and return a new session.
```

Comment:
We do not have to think about that right now but one day elements of a session could not be (L)arrays.
Here, you could replace elements by arrays, no?